libs/custom-scanner/src/custom_a4_scanner.ts

```diff
@@ -0,0 +1,632 @@
+          getImagePortionBySideResult.ok(),
+        ]);
+        scannerImage.imageWidth = imageWidth;
+        scannerImage.imageHeight = imageHeight;
+
+        return ok();
+      }
+    };
+
+    async function processImageData(
+      currentSide: ScanSide,
+      {
+        pageSize,
+        imageWidth,
+        imageHeight,
+      }: { pageSize: number; imageWidth: number; imageHeight: number }
+    ): Promise<Result<void, ErrorCode>> {
+      if (pageSize !== 0 && imageWidth !== 0) {
+        debug(
+          'side %s has data: pageSize=%d, imageWidth=%d, imageHeight=%d',
+          ScanSide[currentSide],
+          pageSize,
+          imageWidth,
+          imageHeight
+        );
+
+        const readImageDataResult = await readImageData(currentSide, {
+          pageSize,
+          imageWidth,
+          imageHeight,
+        });
+
+        /* istanbul ignore next */
+        if (readImageDataResult.isErr()) {
+          return readImageDataResult;
+        }
+      }
+      return ok();
+    }
+
+    const scanLoopTick = async (): Promise<
+      Result<'continue' | 'break', ErrorCode>
+    > => {
+      debug('scan: getting status');
+      const getStatusInternalResult = await this.getStatusInternal();
+
+      if (getStatusInternalResult.isErr()) {
+        errorCount += 1;
+        if (errorCount < maxRetries) {
+          await sleep(1);
+          return ok('continue');
+        }
+
+        return getStatusInternalResult;
+      }
+
+      const { status, a4Status } = convertFromInternalStatus(
+        getStatusInternalResult.ok()
+      );
+      debug('status: %O', status);
+      debug('a4Status: %O', a4Status);
+
+      if (status.isScanCanceled) {
+        debug('scan canceled');
+        return err(ErrorCode.NoDocumentToBeScanned);
+      }
+
+      if (status.isJamPaperHeldBack) {
+        debug('jam paper held back');
+        void (await this.stopScanInternal());
+        return err(ErrorCode.PaperHeldBack);
+      }
+
+      if (status.isPaperJam) {
+        debug('paper jam');
+        void (await this.stopScanInternal());
+        return err(ErrorCode.PaperJam);
+      }
+
+      if (!status.isMotorOn && !status.isScanInProgress) {
+        // The motor is off and the scan is not in progress, maybe an error?
+
+        if (startNoMoveNoScan === 0) {
+          debug('starting to wait for motor on and scan in progress');
+          startNoMoveNoScan = Date.now();
+        } else if (Date.now() - startNoMoveNoScan > maxTimeoutNoMoveNoScan) {
+          debug('waiting for motor on and scan in progress timed out');
+          void (await this.stopScanInternal());
+          return err(ErrorCode.ScannerError);
+        } else {
+          debug('still waiting for motor on and scan in progress');
+        }
+      } else {
+        startNoMoveNoScan = 0;
+      }
+
+      if (
+        ((scanSide & ScanSide.A) === 0 || a4Status.endScanSideA) &&
+        ((scanSide & ScanSide.B) === 0 || a4Status.endScanSideB)
```

Comment:
Why do we use `(scanSide & scanSide.A) === 0` here but use `scanSide === ScanSide.A` below on L464? I'm not sure I understand what these different cases are checking

Reply:
As you might have guessed by now: because that's what their example code did. I'll update them to be uniform.